### PR TITLE
Docs: kpt migration documentation

### PR DESCRIPTION
**Pull request by phanimarupaka**
Here is the migration documentation for kpt 1.0.

### CODE REVIEW

File: site/migration/01-migration.md [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+1. `pkg:` Commands for composing and describing packages.
+2. `fn:` Commands for running functions for validation and customization.
+3. `live:` Commands for interacting with a live cluster.
+
+`PKG_PATH vs DIR:` Most of the commands accept only `PKG_PATH` as input which means
+the input directory must have a valid Kptfile. Few commands can work on just the
+simple `DIR` with resources. Commands operate on the current working directory by default.
+
+| [Existing Commands]                                                                              | [New Commands]                                                                                                                                                                                                                                   |
```
Comment (mikebz) [inline site/migration/01-migration.md:55]: I'd just call this Alpha1 Commands

File: site/migration/01-migration.md [hunk 2]
```diff
@@ -0,0 +1,257 @@
...
+| kpt pkg update LOCAL_PKG_DIR[@VERSION] [flags]                                                   | kpt pkg update [PKG_PATH][@version] [flags] <br> Pulls in upstream changes and merges them into a local package.                                                                                                                                 |
+| kpt pkg fix DIR [flags]                                                                          | kpt fn eval --image gcr.io/kpt-fn/fix:v0.1 <br> Fix a local package which is using deprecated features.                                                                                                                                          |
+| kpt pkg desc DIR/ [flags]                                                                        | Deprecated: read Kptfile directly                                                                                                                                                                                                                |
```
Comment (mikebz) [inline site/migration/01-migration.md:61]: slight language change "Deprecated in favor of reading Kptfile directly" seems less of a command.

File: site/migration/01-migration.md [hunk 3]
```diff
@@ -0,0 +1,257 @@
...
+3. The `OpenAPI` section in `v1alpha1` Kptfile is deprecated. [Details below](#changes-to-setters).
+4. `dependencies` section in `v1alpha1` Kptfile is deprecated. [Details below](#kpt-pkg-sync-deprecated).
+5. `functions` section in `v1alpha1` Kptfile holds definitions for Starlark functions only.
+   This section is deprecated and all the functions can be declared in the `pipeline`
+   section including [Starlark functions].
+6. `inventory-object` is moved to `inventory` section in `v1alpha2` Kptfile. [Details below](#changes-to-live).
+
+### `kpt pkg sync` deprecated
+
+Nothing to worry about. `sync` is just a declarative version of `update` functionality.
```
Comment (mikebz) [inline site/migration/01-migration.md:104]: Wording suggestion:
'sync' functionality has been moved to the 'update` command.

File: site/migration/01-migration.md [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+```
+
+#### Changes made by the function
+
+1. Best effort is made by the function to transform the `packageMetaData` section to
+   the `info` section.
+2. `upstream` section, in the `v1alpha1` Kptfile is converted to `upstream` and `upstreamLock`
+   sections in `v1alpha2` version of Kptfile.
+3. `dependencies` section is removed from Kptfile.
+4. Starlark functions section is removed from Kptfile.
```
Comment (frankfarzan) [inline site/migration/01-migration.md:193]: do you add the starlark function instead?

File: site/migration/01-migration.md [hunk 5]
```diff
@@ -0,0 +1,257 @@
+## Migration from kpt `v0.39.0` to kpt `v1.0`
```
Comment (frankfarzan) [inline site/migration/01-migration.md:1]: It may be better to have this document live in `docs/` instead of `site/`. We can link to this doc from installation page and the fix command instead of having a new top-level section in the navbar.

File: site/migration/01-migration.md [hunk 6]
```diff
@@ -0,0 +1,257 @@
...
+| kpt live init DIR/ [flags]                                                                       | kpt live init [PKG_PATH] [flags] <br> Initializes the package with the name, namespace and id of the resource that will keep track of the package inventory.                                                                                     |
+| kpt live apply DIR/STDIN [flags]                                                                 | kpt live apply [PKG_PATH / -] [flags] <br> Creates, updates and deletes resources in the cluster to make the remote cluster resources match the local package configuration.                                                                     |
+| kpt live preview DIR/STDIN [flags]                                                               | kpt live preview [PKG_PATH / -] [flags] <br> Preview the changes being applied to the live cluster.                                                                                                                                              |
```
Comment (frankfarzan) [inline site/migration/01-migration.md:72]: preview and diff have been removed as well.

File: site/migration/01-migration.md [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+   the desired version of the package, `upstreamLock` is the resolved package state
+   and should not be modified by users.
+3. The `OpenAPI` section in `v1alpha1` Kptfile is deprecated. [Details below](#changes-to-setters).
+4. `dependencies` section in `v1alpha1` Kptfile is deprecated. [Details below](#kpt-pkg-sync-deprecated).
+5. `functions` section in `v1alpha1` Kptfile holds definitions for Starlark functions only.
+   This section is deprecated and all the functions can be declared in the `pipeline`
+   section including [Starlark functions].
+6. `inventory-object` is moved to `inventory` section in `v1alpha2` Kptfile. [Details below](#changes-to-live).
+
+### `kpt pkg sync` deprecated
```
Comment (frankfarzan) [inline site/migration/01-migration.md:102]: To better organize the content to make it easier to navigate. I suggest having the command groups as headers (e.g. `pkg`, `cfg`, `fn`, `live`). ToC should mirror the the command hierarchy accordingly.

File: site/migration/01-migration.md [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+Finally, test your migrated kpt package end-to-end and make sure that the functionality
+is as expected. `gcr.io/kpt-fn/fix` is a helper for migration and doesn't guarantee
+functional parity.
+
+## Timeline
+
+1. `Right now:` You can install and try the pre-release version of kpt `v1.0` binary by following the documentation.
+2. `June 1, 2021:` `gcr.io/kpt-fn/fix` function will be released in kpt-functions-catalog.
+   You can start migrating your existing kpt packages using the function.
+3. `July 1, 2021:` kpt `v1.0` binary will be released. The existing kpt packages are not
```
Comment (frankfarzan) [inline site/migration/01-migration.md:237]: I would reword this: The package format schema will be granduated to v1.
The semver of the CLI binary can be orthogonal to this. In other words, we can guarantee backwards compatibility of package format earlier than 1.0 release of the CLI.

File: site/migration/01-migration.md [hunk 9]
```diff
@@ -0,0 +1,257 @@
...
+an easy way to achieve similar functionality retaining the simplicity of setters,
+which is scoped for post `v1.0` release. Stay tuned.
+
+#### Auto-setters deprecated
+
+[Auto-setters] feature is deprecated in `v1.0` version of kpt. Since the setters are
+migrated to a new and simple declarative version, package consumers can easily
+declare all the setter values and render them all at once.
+
+### Changes to Live
```
Comment (frankfarzan) [inline site/migration/01-migration.md:165]: In each section refer to relevant sections of kpt.dev (reference or book) for more information.

**Review by frankfarzan — COMMENTED** [review]
Please wait for @droot and @mortent to review as well.

File: site/migration/01-migration.md [hunk 10]
```diff
@@ -0,0 +1,257 @@
...
+4. `dependencies` section in `v1alpha1` Kptfile is deprecated. [Details below](#kpt-pkg-sync-deprecated).
+5. `functions` section in `v1alpha1` Kptfile holds definitions for Starlark functions only.
+   This section is deprecated and all the functions can be declared in the `pipeline`
+   section including [Starlark functions].
+6. `inventory-object` is moved to `inventory` section in `v1alpha2` Kptfile. [Details below](#changes-to-live).
+
+### `kpt pkg sync` deprecated
+
+Nothing to worry about. `sync` is just a declarative version of `update` functionality.
+The functionality is offered by `kpt pkg update` in a more user friendly fashion.
```
Comment (frankfarzan) [inline site/migration/01-migration.md:105]: You may want to address the fact the Kptfile doesn't have `dependencies` section. What's the equivalent flow?

File: site/migration/01-migration.md [hunk 11]
```diff
@@ -0,0 +1,257 @@
...
+- [Timeline](#Timeline)
+
+kpt `v1.0` is going to be the latest major release of the kpt CLI. The implementation of kpt `v1.0`
+has changed considerably from kpt `v0.39.0`. A rich set of new features have been added
+as a result of the customers’ input and requirements. Some features have been
+deprecated or refactored in ways that make them incompatible with kpt `v0.39.0`.
+Since these are backwards incompatible changes, there should be a way for customers
+to migrate/fix their existing kpt packages which are compatible with `v0.39.0` version of kpt,
+to become compatible with kpt `v1.0`. This document outlines the end to end migration
+journey of customers using a comprehensive kpt package example.
```
Comment (frankfarzan) [inline site/migration/01-migration.md:29]: Somewhere at the top, link to the kpt book. Specifically, the user needs to read Chapter 1 and 2, i.e. the Quickstart and the concepts, specifically the workflows (https://kpt.dev/book/02-concepts/02-workflows) to be able to make sense of things. Trying to follow the migration doc (dealing with complexity of two versions) without understanding the basic middle of new version will lead to confusion.

File: site/migration/01-migration.md (hunk 10 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md:105]: We need to have a separate doc for that. Once that is ready we can link it here https://github.com/GoogleContainerTools/kpt/issues/1941

File: site/migration/01-migration.md (hunk 1 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Used version numbers instead as we are using it all across the docs. Users might not have observed v1alpha1 in Kptfile at all.

File: site/migration/01-migration.md (hunk 2 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Update.

File: site/migration/01-migration.md (hunk 3 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md:113]: Good point. Update to `sync` merged with `update`

File: site/migration/01-migration.md (hunk 4 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md:204]: No. Added it to manual portion.

File: site/migration/01-migration.md (hunk 6 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Updated.

File: site/migration/01-migration.md (hunk 5 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md:1]: Will do that in next PR.

File: site/migration/01-migration.md (hunk 7 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Updated.

File: site/migration/01-migration.md (hunk 8 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md:248]: I feel the wording you suggested is complex. I am having trouble understanding it.

File: site/migration/01-migration.md (hunk 9 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md]: Updated.

File: site/migration/01-migration.md (hunk 11 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md:31]: Updated Before you begin section.

**Comment by phanimarupaka** [comment]
@frankfarzan apologies for force push. Did by mistake.

**Review by mikebz — APPROVED** [review]
let's merge for fishfood.

File: site/migration/01-migration.md (hunk 8 above)
Comment (frankfarzan) [reply site/migration/01-migration.md:248]: There are two separate independent releases: The package format (ie.g. Kptfile and other CRD) can be graduated to apiVersion v1 *before* we release CLI version 1.0.

File: site/migration/01-migration.md (hunk 5 above)
Comment (frankfarzan) [reply site/migration/01-migration.md:1]: Let's do this in this PR, otherwise the navbar will be affected in the meantime.

File: site/migration/01-migration.md [hunk 12]
```diff
@@ -0,0 +1,278 @@
+## Migration from kpt `v0.39.0` to kpt `v1.0`
+
+- [Before you begin](#Before-you-begin)
```
Comment (frankfarzan) [inline site/migration/01-migration.md:3]: In the GitHub preview, these don't show up as links. Not sure if that's a real issue or not.

**Review by frankfarzan — COMMENTED** [review]
Looking good. I like the organization now.

File: site/migration/01-migration.md (hunk 8 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md:248]: So we have different timeline for both ? Can you make please a separate PR using edit icon after this PR is merged?

File: site/migration/01-migration.md (hunk 5 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md]: It is not possible to point to files outside the site folder from the files within site folder. So moved the folder to installation folder and added a link in installation guide. Remove it from sidebar to not pollute the view.

File: site/migration/01-migration.md (hunk 12 above)
Comment (phanimarupaka) [reply site/migration/01-migration.md]: They do for me. Try this out https://github.com/GoogleContainerTools/kpt/blob/51e2495577f27c272210dcd8a17f3090797f2f91/site/installation/migration/01-migration.md#Before-you-begin

File: site/installation/README.md [hunk 13]
```diff
@@ -1,5 +1,7 @@
 Users can get kpt in a variety of ways:
 
+**Note:** Please follow the [migration guide] to make your existing kpt packages compatible
```
Comment (frankfarzan) [inline site/installation/README.md:3]: To future proof this, reword:

If you're currently using version v0.39, follow the ... to migrate.

File: site/installation/README.md [hunk 14]
```diff
@@ -95,3 +97,4 @@ kpt help
 [cloud-sdk]: https://github.com/GoogleCloudPlatform/cloud-sdk-docker
 [linux]: https://github.com/GoogleContainerTools/kpt/releases/download/v1.0.0-alpha.4/kpt_linux_amd64
 [darwin]: https://github.com/GoogleContainerTools/kpt/releases/download/v1.0.0-alpha.4/kpt_darwin_amd64
+[migration guide]: installation/migration/01-migration.md
```
Comment (frankfarzan) [inline site/installation/README.md:100]: you can move to /docs and then link using

https://github.com/GoogleContainerTools/kpt/blob/next/docs/migration.md

File: site/installation/migration/01-migration.md [hunk 15]
```diff
@@ -0,0 +1,278 @@
+## Migration from kpt `v0.39.0` to kpt `v1.0`
```
Comment (frankfarzan) [inline site/installation/migration/01-migration.md:1]: if you're keeping in /site not sure why it can't be in `installation/migration.md`

**Review by frankfarzan — APPROVED** [review]
/lgtm with some comments.
